### ISSUE TITLE
Outliner - Filter Panel - Refactor how indentation is handled

### ISSUE THREAD

**Issue by Quackarooni** [Bforartists/Bforartists#4627]
| User Interface | Source Code |
|:---:|:---:|
|![image](https://github.com/user-attachments/assets/ce7e3492-a327-43e7-947f-6050950e63d7)|![image](https://github.com/user-attachments/assets/9ebd8410-c912-4b5d-bf0d-d2dec3b51048)|

Currently, indentation in the Outliner's Filter Panel is handled through adding separators as spacers for each indented element. While sufficient, it having to be applied per row comes with the likelihood of forgetting to do it. 

This comes at the risk of bugs when merging changes from upstream, as this indentation is BFA-only.

The plan is to make this less likely by using an indentation system based on `layout.split` instead. That way, the indentation would be defined once per sublayout, and any amount of rows could be added this sublayout without having to specify indentation.

**Comment by ReinerBforartists** [comment]
I have merged your pull request so that you can start from there and don't run into a merge conflict with your own changes here :)

**Comment by Quackarooni** [comment]
> May i point at one of the caveats here already. Generally spoken, since here it is not really a problem. Split layout has not really a control over the width. It is in per cent. And changes with the width of the panel then.

Yeah, am aware. Still mulling over as to whether I should make a helper function to use `split` but with absolute width instead of relative. `rna_keymap_ui` does something like that with `def _indented_layout`.

Though prolly should say I'm not attached on having to use `layout.split` for this. 
If it proves too cumbersome or not that helpful, I'm fine with leaving the implementation.

Just moreso wanted to take a crack at this and see whether the implementation could be improved or not.

**Comment by ReinerBforartists** [comment]
Yeah, that's great. I look forward to what you come up with :)

**Comment by Quackarooni** [comment]
So I'm prolly not gonna radically change the code here like I initially planned, since I don't see the need to do so.

I'll just do a cleanup pass on the existing code, since the messiness in the existing code is what initially bothered me. 
After that gets merged, we can prolly close this issue.
